modules/generative-cohere/clients/cohere.go

```diff
@@ -189,22 +190,23 @@ func (v *cohere) getApiKey(ctx context.Context) (string, error) {
 }
 
 type generateInput struct {
-	Prompt            string   `json:"prompt"`
-	Model             string   `json:"model"`
-	MaxTokens         int      `json:"max_tokens"`
-	Temperature       int      `json:"temperature"`
-	K                 int      `json:"k"`
-	StopSequences     []string `json:"stop_sequences"`
-	ReturnLikelihoods string   `json:"return_likelihoods"`
+	ChatHistory   []message `json:"chat_history"`
+	Message       string    `json:"message"`
+	Model         string    `json:"model"`
+	MaxTokens     int       `json:"max_tokens"`
+	Temperature   int       `json:"temperature"`
+	K             int       `json:"k"`
+	StopSequences []string  `json:"stop_sequences"`
 }
 
-type generateResponse struct {
-	Generations []generation
-	Error       *cohereApiError `json:"error,omitempty"`
+type message struct {
+	Role    string `json:"role"`
```

Comment:
you can make an enum type for role like:
```
type role string

var (
   roleChatBot role = "CHATBOT"
   ...
)

type message struct {
   Role role `json:"role"`
}
```

Reply:
Apologies I don't understand this since we aren't passing in the message, I get `var roleChatBot unused`